test/e2e/common_test.go

```diff
@@ -479,19 +482,23 @@ func GetPortLock(port string) *lockfile.LockFile {
 	return lock
 }
 
-// GetRandomIPAddress returns a random IP address to avoid IP
-// collisions during parallel tests
-func GetRandomIPAddress() string {
-	// To avoid IP collisions of initialize random seed for random IP addresses
-	rng := rand.New(rand.NewSource(time.Now().UnixNano()))
-
-	nProcs := GinkgoT().ParallelTotal()
-	myProc := GinkgoT().ParallelProcess() - 1
-
-	// 10.88.255 is unlikely to be used by CNI or netavark.
-	// Last octet .0 - .254, and will be unique to this ginkgo process.
-	ip4 := strconv.Itoa(rng.Intn((255-nProcs)/nProcs)*nProcs + myProc)
-	return "10.88.255." + ip4
+// GetSafeIPAddress returns a sequentially allocated IP address that _should_
+// be safe and unique across parallel tasks
+//
+// Used by tests which want to use "--ip SOMETHING-SAFE". Picking at random
+// just doesn't work: we get occasional collisions. Our current approach
+// allocates a /24 subnet for each ginkgo process, starting at .128.x, see
+// BeforeEach() above. Unfortunately, CNI remembers each address assigned
+// and assigns <previous+1> by default -- so other parallel jobs may
+// get IPs in our block. The +10 leaves a gap for that. (Netavark works
+// differently, allocating sequentially from .0.0, hence our .128.x).
+// This heuristic will fail if run in parallel on >127 processors or if
+// one test calls us more than 25 times or if some other test runs more
+// than ten networked containers at the same time as any test that
+// relies on GetSafeIPAddress(). I'm finding it hard to care.
+func GetSafeIPAddress(base *int) string {
+	*base += 10
+	return fmt.Sprintf("10.88.%d.%d", *base/256, *base%256)
```

Comment:
```suggestion
func GetSafeIPAddress(base *int) string {
	*base += 10
	return fmt.Sprintf("10.88.%d.%d", *base/255, *base%255)
```
Note sure if this has other side effects but you must make sure the third and fourth octet can never be 255 at the same time. 10.88.255.255 is the broadcast address and cannot be assigned to containers.

Reply:
octet 3 will be 255 iff running on ginkgo processor 127. I'm choosing to stick my head in the sand about that.

octet 4 ... its initial value is 2, and gets incremented by 10, so I'm thinking it's unlikely ever to be 255. That's specifically why I chose 256, it's a cheap way to do two octets. (Side note: I initially used uint16 but gave up after ending up with waaaaay too many cluttery typecasts). Have I missed something? I intend to keep the code as 256, but will wait to re-push; please lmk if there's something I missed.

Reply:
Yeah you make a fair point, if you think this cannot happen I am fine with it. It is to late for me to do this math in my head.

Reply:
Understood. There is no rush on this, it can wait for tomorrow or later. I've been testing heavily on my no-flake-retry PR, and (so far) no dup-IP failures, compared to at least one per run since merging #18858. My latest test run has 15 minutes to go, I'll let that finish before re-pushing this.

(Side note: "unlikely", in my comment above, was a euphemism for "impossible under the rules of the universe I live in except for cosmic-ray glitches": even number, plus `N` * another even number, can never be odd).